File: SecondoAnno/ASD/Modulo1/Algoritmi/Ordinamento/SelectionSort.py
def selection_sort(l):
    if len(l) <= 1:
        return l
    
    for k in range (len(l)):
        i = k+1
        for i in range(len(l)):
            if l[i] > l[k]:
                l[i], l[k] = l[k], l[i]
    return l

l = [4,8,9,6,5,7,2,5,4,6,1]
print(selection_sort(l))

File: TerzoAnno/IS/EsempioStrategy/SelectionSort.java
import java.util.List;

public class SelectionSort implements SortAlgorithm {
    public List sort(List unSortedList) {
        System.out.println("SelectionSort");
        for (int i = 0; i < unSortedList.size(); i++) {
            int minIndex = i;
            for (int j = i + 1; j < unSortedList.size(); j++) {
                if ((int) unSortedList.get(j) < (int) unSortedList.get(minIndex)) {
                    minIndex = j;
                }
            }
            int temp = (int) unSortedList.get(i);
            unSortedList.set(i, unSortedList.get(minIndex));
            unSortedList.set(minIndex, temp);
        }
        return unSortedList;
    }
    
}
